core/app/models/spree/payment.rb

```diff
@@ -37,6 +37,8 @@ class Payment < Spree.base_class
 
     before_validation :validate_source
 
+    after_initialize :set_amount, if: -> { new_record? && order.present? && !amount_changed? }
```

Comment:
@damianlegawiec We'll be setting the payment amount here if no custom amount is provided